projectile.el

```diff
@@ -867,16 +867,31 @@ Files are returned as relative paths to the project root."
      ((eq vcs 'git) projectile-git-submodule-command)
      (t ""))))
 
-(defun projectile-get-all-sub-projects (project)
-  "Get all sub-projects for a given projects.
-PROJECT is base directory to start search recursively."
+(defun projectile-get-immediate-sub-projects (project)
+  "Get immediate sub-projects for a given project. Does not recurse.
+It is assumed that `project` will end with an appropriate path delimiter, such as '/' or a '\'"
   (let* ((default-directory project)
          ;; search for sub-projects under current project `project'
          (submodules (mapcar
                       (lambda (s)
-                        (file-name-as-directory (expand-file-name s default-directory)))
-                      (projectile-files-via-ext-command (projectile-get-sub-projects-command)))))
+                        (file-name-as-directory (expand-file-name s project)))
+                      (projectile-files-via-ext-command (projectile-get-sub-projects-command))))
+         (project-child-folder-regex
+          (concat "^"
+                  (regexp-quote default-directory))))
+
+    ;; If project root is inside of an SCM folder, but not actually an
+    ;; SCM root itself, external paths will leak in. Let's remove them
+    (-filter (lambda (path)
+               (string-match-p project-child-folder-regex
+                               path))
+             submodules)))
+
+(defun projectile-get-all-sub-projects (project)
```

Comment:
This should come before the other definition to avoid a compiler warning about using something that's not yet defined.